xchange-stream-btcmarkets/src/main/java/info/bitrich/xchangestream/btcmarkets/BTCMarketsStreamingService.java

```diff
@@ -13,57 +13,78 @@
 
 class BTCMarketsStreamingService extends JsonNettyStreamingService {
   static final String CHANNEL_ORDERBOOK = "orderbook";
+  static final String CHANNEL_TICK = "tick";
   static final String CHANNEL_HEARTBEAT = "heartbeat";
   private static final Logger LOG = LoggerFactory.getLogger(BTCMarketsStreamingService.class);
   private final Set<String> subscribedOrderbooks = Sets.newConcurrentHashSet();
+  private final Set<String> subscribedtick = Sets.newConcurrentHashSet();
 
   public BTCMarketsStreamingService(String apiUrl) {
     super(apiUrl);
   }
 
   private BTCMarketsWebSocketSubscribeMessage buildSubscribeMessage() {
     return new BTCMarketsWebSocketSubscribeMessage(
-        new ArrayList<>(subscribedOrderbooks),
-        Lists.newArrayList(CHANNEL_ORDERBOOK, CHANNEL_HEARTBEAT),
-        null,
-        null,
-        null);
+            new ArrayList<>(subscribedtick),
+            Lists.newArrayList(CHANNEL_ORDERBOOK,CHANNEL_HEARTBEAT,CHANNEL_TICK),
+            null,
+            null,
+            null);
   }
 
   @Override
   protected String getChannelNameFromMessage(JsonNode message) {
     final String messageType = message.get("messageType").asText();
-    if (messageType.startsWith(CHANNEL_ORDERBOOK)) {
+    if (messageType.startsWith(CHANNEL_TICK)) {
       return messageType + ":" + message.get("marketId").asText();
     }
-    return messageType;
+    else  if (messageType.startsWith(CHANNEL_ORDERBOOK)) {
+      return messageType + ":" + message.get("marketId").asText();
+    }
+    else
+      return messageType;
   }
 
   @Override
   public String getSubscribeMessage(String channelName, Object... args) throws IOException {
```

Comment:
can we add a sanity check here to make sure arg[0] is a string and log if its not, will make debugging easier

Reply:
I don't think there is a need for it, `messageType` is already declared as a String Type, so test will always be true. Message type returned by [BTCMarkets](https://github.com/BTCMarkets/API/wiki/WebSocket-v2) is always a string also. Am I missing Something?

Reply:
This is not for `messageType` I'm talking about `getSubscribeMessage()`

Reply:
Fixed, Unless I got it wrong

Reply:
> 
> 
> I don't see any tests here.

I haven't done any test cases yet. I just took the library as it is and made some changes to add **Ticker**. Existing manual example was just enough for our use case. My comment in the first commit might be some misleading, I just tested cases in manual example, and it worked for our purpose.
I'm busy with another project, might be later I could revisit. Feel free to make any changes if you feel like. I might be stuck with my work for a while.